### ISSUE TITLE
Can't run the application

### ISSUE THREAD

**Issue by igungor** [codito/gnome-pomodoro#266]
I'm using Arch Linux, gnome-shell 3.22.2.

I've installed the extension from the AUR package here: https://aur.archlinux.org/packages/gnome-shell-pomodoro/

Running gnome-pomodoro from command line returns this:

```
~  Δ gnome-pomodoro 
**
ERROR:gnome.c:315:gnome_extension_state_to_string: code should not be reached
zsh: abort (core dumped)  gnome-pomodoro
~  Δ 
```

Then I cloned https://github.com/codito/gnome-pomodoro/tree/gnome-3.22 and installed it with `./autogen.sh --enable-debug` and this is the coredump:

```
~  Δ cat coredump                                                        
           PID: 3783 (gnome-pomodoro)
           UID: 1000 (ibrahim)
           GID: 100 (users)
        Signal: 6 (ABRT)
     Timestamp: Wed 2017-02-08 16:51:12 +03 (1h 13min ago)
  Command Line: /usr/bin/gnome-pomodoro --no-default-window
    Executable: /usr/bin/gnome-pomodoro
 Control Group: /user.slice/user-1000.slice/user@1000.service/dbus.service
          Unit: user@1000.service
     User Unit: dbus.service
         Slice: user-1000.slice
     Owner UID: 1000 (ibrahim)
       Boot ID: f1c9fdc56eca4b5684a0a84e5196330c
    Machine ID: 5f66538763134959be8f5b42cdb6c780
      Hostname: zen
       Storage: /var/lib/systemd/coredump/core.gnome-pomodoro.1000.f1c9fdc56eca4b5684a0a84e5196330c.3783.1486561872000000000000.lz4
       Message: Process 3783 (gnome-pomodoro) of user 1000 dumped core.
                
                Stack trace of thread 3783:
                #0  0x00007ff30246604f raise (libc.so.6)
                #1  0x00007ff30246747a abort (libc.so.6)
                #2  0x00007ff302a5f2a5 g_assertion_message (libglib-2.0.so.0)
                #3  0x00007ff302a5f33a g_assertion_message_expr (libglib-2.0.so.0)
                #4  0x00007ff2e0c9b0b1 gnome_extension_state_to_string (libgnome.so)
                #5  0x00007ff2e0ca00a3 n/a (libgnome.so)
                #6  0x00007ff302d10f75 g_closure_invoke (libgobject-2.0.so.0)
                #7  0x00007ff302d22f82 n/a (libgobject-2.0.so.0)
                #8  0x00007ff302d2bbcc g_signal_emit_valist (libgobject-2.0.so.0)
                #9  0x00007ff302d2c45b g_signal_emit_by_name (libgobject-2.0.so.0)
                #10 0x00007ff2e0c9aa52 n/a (libgnome.so)
                #11 0x00007ff2ff8801c8 ffi_call_unix64 (libffi.so.6)
                #12 0x00007ff2ff87fc2a ffi_call (libffi.so.6)
                #13 0x00007ff302d117ae g_cclosure_marshal_generic (libgobject-2.0.so.0)
                #14 0x00007ff302d10f75 g_closure_invoke (libgobject-2.0.so.0)
                #15 0x00007ff302d2337d n/a (libgobject-2.0.so.0)
                #16 0x00007ff302d2bbcc g_signal_emit_valist (libgobject-2.0.so.0)
                #17 0x00007ff302d2bfaf g_signal_emit (libgobject-2.0.so.0)
                #18 0x00007ff30301a85c n/a (libgio-2.0.so.0)
                #19 0x00007ff30300a344 n/a (libgio-2.0.so.0)
                #20 0x00007ff302a3843a g_main_context_dispatch (libglib-2.0.so.0)
                #21 0x00007ff302a387f0 n/a (libglib-2.0.so.0)
                #22 0x00007ff302a3889c g_main_context_iteration (libglib-2.0.so.0)
                #23 0x00007ff302ff254d g_application_run (libgio-2.0.so.0)
                #24 0x0000000000400c31 _vala_main (gnome-pomodoro)
                #25 0x00007ff302453291 __libc_start_main (libc.so.6)
                #26 0x0000000000400a7a _start (gnome-pomodoro)
                
                Stack trace of thread 3789:
                #0  0x00007ff30251248d poll (libc.so.6)
                #1  0x00007ff2e6b0bee1 n/a (libpulse.so.0)
                #2  0x00007ff2e6afd6f1 pa_mainloop_poll (libpulse.so.0)
                #3  0x00007ff2e6afdd8e pa_mainloop_iterate (libpulse.so.0)
                #4  0x00007ff2e6afde40 pa_mainloop_run (libpulse.so.0)
                #5  0x00007ff2e6b0be29 n/a (libpulse.so.0)
                #6  0x00007ff2e68a9fe8 n/a (libpulsecommon-10.0.so)
                #7  0x00007ff3027d8454 start_thread (libpthread.so.0)
                #8  0x00007ff30251b7df __clone (libc.so.6)
                
                Stack trace of thread 3786:
                #0  0x00007ff30251248d poll (libc.so.6)
                #1  0x00007ff302a38786 n/a (libglib-2.0.so.0)
                #2  0x00007ff302a3889c g_main_context_iteration (libglib-2.0.so.0)
                #3  0x00007ff2f4f614bd n/a (libdconfsettings.so)
                #4  0x00007ff302a600d5 n/a (libglib-2.0.so.0)
                #5  0x00007ff3027d8454 start_thread (libpthread.so.0)
                #6  0x00007ff30251b7df __clone (libc.so.6)
                
                Stack trace of thread 3784:
                #0  0x00007ff30251248d poll (libc.so.6)
                #1  0x00007ff302a38786 n/a (libglib-2.0.so.0)
                #2  0x00007ff302a3889c g_main_context_iteration (libglib-2.0.so.0)
                #3  0x00007ff302a388e1 n/a (libglib-2.0.so.0)
                #4  0x00007ff302a600d5 n/a (libglib-2.0.so.0)
                #5  0x00007ff3027d8454 start_thread (libpthread.so.0)
                #6  0x00007ff30251b7df __clone (libc.so.6)
                
                Stack trace of thread 3785:
                #0  0x00007ff30251248d poll (libc.so.6)
                #1  0x00007ff302a38786 n/a (libglib-2.0.so.0)
                #2  0x00007ff302a38b12 g_main_loop_run (libglib-2.0.so.0)
                #3  0x00007ff30301e316 n/a (libgio-2.0.so.0)
                #4  0x00007ff302a600d5 n/a (libglib-2.0.so.0)
                #5  0x00007ff3027d8454 start_thread (libpthread.so.0)
                #6  0x00007ff30251b7df __clone (libc.so.6)
                
                Stack trace of thread 3788:
                #0  0x00007ff30251248d poll (libc.so.6)
                #1  0x00007ff2e6b0bee1 n/a (libpulse.so.0)
                #2  0x00007ff2e6afd6f1 pa_mainloop_poll (libpulse.so.0)
                #3  0x00007ff2e6afdd8e pa_mainloop_iterate (libpulse.so.0)
                #4  0x00007ff2e6afde40 pa_mainloop_run (libpulse.so.0)
                #5  0x00007ff2e6b0be29 n/a (libpulse.so.0)
                #6  0x00007ff2e68a9fe8 n/a (libpulsecommon-10.0.so)
                #7  0x00007ff3027d8454 start_thread (libpthread.so.0)
                #8  0x00007ff30251b7df __clone (libc.so.6)
                
                Stack trace of thread 3790:
                #0  0x00007ff302516f19 syscall (libc.so.6)
                #1  0x00007ff302a7e03a g_cond_wait_until (libglib-2.0.so.0)
                #2  0x00007ff302a0ce89 n/a (libglib-2.0.so.0)
                #3  0x00007ff302a0d4ac g_async_queue_timeout_pop (libglib-2.0.so.0)
                #4  0x00007ff302a60b9d n/a (libglib-2.0.so.0)
                #5  0x00007ff302a600d5 n/a (libglib-2.0.so.0)
                #6  0x00007ff3027d8454 start_thread (libpthread.so.0)
                #7  0x00007ff30251b7df __clone (libc.so.6)
```

Please let me know if anything else is necessary.

**Comment by mbooth101** [comment]
Hi,

It looks like Fedora users are seeing the same/similar problem: https://bugzilla.redhat.com/show_bug.cgi?id=1422595

Full backtrace/thread dump can be seen here: https://bugzilla.redhat.com/attachment.cgi?id=1250651

**Comment by kamilprusko** [comment]
Thanks for the report.

fixed in 8eef8c1541de801696c7615c993586384636068d

Bug affects first-time users mostly - pomodoro would work on next login / gnome-shell restart.

**Comment by demizer** [comment]
> Bug affects first-time users mostly
This was my experience. After reboot it loaded fine.
